### PR TITLE
Propose change from review of the guide

### PR DESCRIPTION
**Pull request by lacasseio**
Here are my proposed changes for the code review of the guide.

I feel there may be an issue if you have complete zero knowledge of Gradle and go through this tutorial. However, we can address this at a later time.

### CODE REVIEW

File: contents/index.adoc [hunk 1]
```diff
@@ -124,7 +124,7 @@ $ ./gradlew greeterStaticLibrary
 BUILD SUCCESSFUL
 
 ----
-<1> Gradle does not show the compiler commands or the compiler output in order to keep the output from Gradle itself clean. But the compiler output can be found in `build/tmp/compileGreeterStaticLibraryGreeterCpp/output.txt` and the compiler flags can be found in `build/tmp/compileGreeterStaticLibraryGreeterCpp/options.txt`
+<1> Gradle does not show the compiler commands or only show an excert of the compiler output if the compilation failed in order to keep the output from Gradle itself clean. But the compiler output can be found in `build/tmp/compileGreeterStaticLibraryGreeterCpp/output.txt` and the compiler flags can be found in `build/tmp/compileGreeterStaticLibraryGreeterCpp/options.txt`
```
Comment (ysb33r) [inline contents/index.adoc]: I assume you meant `excerpt` ?

File: src/greeter/cpp/greeter.cpp [hunk 2]
```diff
@@ -1,7 +1,8 @@
 #include "greeter.hpp"
+#include <iostream>
```
Comment (ysb33r) [inline src/greeter/cpp/greeter.cpp]: We shoul dprobably swap the two headers around as per convention in the C++ world.

Comment (lacasseio) [inline src/greeter/cpp/greeter.cpp]: The [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) has a good description of the `#include` ordering. The prototype header first is what I have seen most often with good reason. I'm open to other convention.

File: contents/index.adoc (hunk 1 above)
Comment (lacasseio) [inline contents/index.adoc]: You are right, thanks. I changed it.

File: src/greeter/cpp/greeter.cpp (hunk 2 above)
Comment (ysb33r) [inline src/greeter/cpp/greeter.cpp]: The opposing view is to order them the other way around as not to allow any local includes to pollute system-wide ones with #define overrides.

It Gradle wants to follow the Google guidelines, then we have to standardize and add a note to the Style Guide.

Comment (lacasseio) [inline src/greeter/cpp/greeter.cpp]: No worries, let's change it now and discuss that in the future.

Comment (benignbala) [inline src/greeter/cpp/greeter.cpp]: Generally, I've been following this order

1. Interface that's being implemented
2. Standard headers 
3. 3rd party header files (boost for eg.)
4. Other necessary header files from the current project.

But my organisation (work place) completely follows the order mentioned by @ysb33r. Apart from the Google's style guide, others don't seem to recommend a style. So, I guess it is good to discuss this have this established (may be separately ?). Thanks

Comment (benignbala) [inline src/greeter/cpp/greeter.cpp]: BTW, in the current case, I'd included `<iostream>` in the `greeter.hpp` file and that's why it wasn't included  the cpp file. Apologies for not sticking to standards/guidelines with respect to the code.

Comment (lacasseio) [inline src/greeter/cpp/greeter.cpp]: @benignbala It's not your fault. We should document this a bit better. We will [discuss this](https://github.com/gradle/gradle-native/issues/35) and come up with guidelines for all our native sample code.

**Comment by ysb33r** [comment]
@benignbala Can you look to see whether you can merge @lacasseio 's recommendations into the current PR you have for the testing?

**Comment by ysb33r** [comment]
@benignbala Has taken care of this on his branch - will appear in final PR

**Pull request closed without merging**